### ISSUE TITLE
Lab1 peer review - Good job!

### ISSUE THREAD

**Issue by simocosimo** [aminmbare/Computational-intelligence#1]
Hi! :)
## Main comment
Nice choice of the algorithm: I like the implementation and the way you prune the search tree to avoid useless computation.
I would say that the biggest advantage of this implementation is the fact that you can find a solution even for high values of N in a reasonable time.

The main con is the fact that you end up exploring a lot of nodes to find the best solution (for N = 20 you explore 101999 nodes with pruning): an A* would go through fewer nodes (mine was around 15k and I didn't do a lot of pruning).

Maybe this approach combined with a heuristic estimation like the one in A* could lead to beautiful results, maintaining both the time advantage of the branch and bound and the node exploration pro of the A*.

## Little things
It's really nothing about the code, just a suggestion to help you organize the repo better: you have the lab1 README.md file in the root dir of your repo (together with the poetry files needed for it).
You can create a 'lab1' folder and put everything in it so that you have a README for each lab (and do this for every lab).

Basically a structure like this
```
/<root-of-repo>
    /lab1
        README.md
        poetry.lock
        /lab1
            lab1.py
    /lab2
    /lab3
```
Just a hint, maybe it helps!
Overall, good job! 👍🏼
